src/libsyntax/ast.rs

```diff
@@ -166,12 +159,15 @@ pub const ILLEGAL_CTXT : SyntaxContext = 1;
            RustcEncodable, RustcDecodable, Clone, Copy)]
 pub struct Name(pub u32);
 
+impl<T: AsRef<str>> PartialEq<T> for Name {
```

Comment:
I figured this would make life a lot easier. sadly I cannot implement the commutative version of this.

The alternative was often `&*ident.name.as_str() == "text"` when it's now `ident.name == "text"`